src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,106 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static int DEFAULT_CAPACITY = 16;
+    private static double LOAD_FACTOR = 0.75;
+    private static int CONSTTANT = 2;
+    private int size;
+    private Node<K,V>[] bucket;
+    private int threshold = 0;
+
+    public MyHashMap() {
+        this.bucket = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resizeIfNedeed();
+        int index = getIndex(key);
+        Node<K, V> node = bucket[index];
+        while (node != null) {
+            if (Objects.equals(key, node.key)) {
+                node.value = value;
+                return;
+            }
+            node = node.next;
+        }
+        if (key == null) {
+            addNewNode(0, new Node<>(key, value));
+            return;
+        }
+        addNewNode(index, new Node<>(key, value));
     }
 
     @Override
     public V getValue(K key) {
+        int hash = getHash(key);
+        int index = hash % bucket.length;
+        Node<K, V> node = bucket[index];
+        Node<K,V> node1 = bucket[0];
+        while (node != null) {
+            if (key != null && Objects.equals(node.key, key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
+        for (int i = 0; i < bucket.length; i++) {
+            if (key == null && Objects.equals(node1.key, key)) {
+                return node1.value;
+            }
+            if (node1 == null) {
+                break;
+            }
+            node1 = node1.next;
+        }
```

Comment:
I dont understand, why do we need the second one for-loop?
node1 is the zero bucket -> any guarantee that here is element
node is the bucket by index -> here element 100% present

Let's leave only one loop.
```suggestion
        while (node != null) {
            if (key != null && Objects.equals(node.key, key)) {
                return node.value;
            }
            node = node.next;
        }
```

Reply:
not fixed